File: src/main/resources/templates/swift/solution/Sources/${packageNameFolder}Lib/Policy.swift
import Foundation

public class Policy {
    private var context: Context!

    init(_ context: Context!) {
        self.context = context
    }

    /// Chooses a strategy depending on the number of date objects.
    public func configure() {
        if self.context.getDates().count > 10 {
            print("-> More than 10 dates, choosing merge sort!")
            self.context.setSortAlgorithm(MergeSort())
        } else {
            print("-> Less or equal than 10 dates. choosing quick sort!")
            self.context.setSortAlgorithm(BubbleSort())
        }
    }
}

File: src/main/resources/templates/c_sharp/solution/Policy.cs
namespace assignment;

public class Policy
{
    private const int DATES_SIZE_THRESHOLD = 10;

    private Context context;

    public Policy(Context context)
    {
        this.context = context;
    }

    public Context Context { get => context; set => context = value; }

    /// <summary>
    /// Chooses a strategy depending on the number of date objects.
    /// </summary>
    public void Configure()
    {
        if (this.context.Dates == null)
        {
            throw new InvalidOperationException("Dates of Context has to be set before Configure() can be called");
        }

        if (this.context.Dates.Count > DATES_SIZE_THRESHOLD)
        {
            this.context.SortAlgorithm = new MergeSort();
        }
        else
        {
            this.context.SortAlgorithm = new BubbleSort();
        }
    }
}
